### PR TITLE
feat: support api keys rotation

### PR DESCRIPTION
**Pull request by a-cordier**
see https://gravitee.atlassian.net/browse/GKO-2561

depends on https://github.com/gravitee-io/gravitee-api-management/pull/16605

## Support custom API keys and key rotation for subscriptions

Add the ability to specify custom API keys on Subscription CRDs and rotate them declaratively.

- Add `apiKeys` field to the subscription spec with `ApiKeySpec` (key + optional expireAt)
- Enforce key length (32-256 chars) via kubebuilder schema validation and key uniqueness / plan type checks in the admission webhook
- Propagate API keys to APIM on create and reconcile key changes on update (add new keys, revoke removed ones)
- Add examples for subscription with custom API keys and secret-based key loading
- Add admission tests (duplicate keys, wrong plan type, invalid expireAt) and integration tests for key rotation (instant and gradual)

More info here https://gravitee.atlassian.net/browse/GKO-2550

### CODE REVIEW

File: internal/admission/subscription/validate.go [hunk 1]
```diff
@@ -247,18 +247,38 @@ func validatePlanSecurityType(plan core.PlanModel, planName string) *errors.Admi
...
+		if seen[k.GetKey()] {
+			return errors.NewSeveref(
+				"duplicate key [%s] in apiKeys",
+				k.GetKey(),
+			)
+		}
+		seen[k.GetKey()] = true
+
+		if k.GetExpireAt() != nil {
+			if err := validateEndingAt(k.GetExpireAt()); err != nil {
```
[the remark this answers is not in the document]
Comment (a-cordier) [reply internal/admission/subscription/validate.go:273]: I don't think it's necessarily a bad thing to fail on apply when an expiryDate is overdue ... It prevents users to silently expire an API key. If they really want to revoke a key, the way to do so is by removal from the list. I would keep it as it and if see the feedbacks we have from there.

**Review by benoitgravitee — APPROVED** [review]
